### DISCUSSION TITLE
Z2T endpoint friendly name?

### DISCUSSION THREAD

**Discussion by urechi** [arendst/Tasmota#16630] [Ideas]
Similar to the device friendly name, for devices with multiple endpoints we could add the ability to set a friendly name for each endpoint. For example, a switch with 3 gangs named `Hall_switch` that controls:
1. Light
2. Fan
3. Garage_Door

`ZbEndpointName<x> <device>,<name_of_endpoint_x>`
```
ZbEndpointName1 0x0000,Light
ZbEndpointName2 0x0000,Fan
ZbEndpointName3 0x0000,Garage_Door
```
OR

`ZbName <device>,<name_of_endpoint_x>,<x>` (without breaking changes):
```
ZbName 0x0000,Light,1
ZbName 0x0000,Fan,2
ZbName 0x0000,Garage_Door,3
```

The `SENSOR` topic with `SetOption89` `SetOption120` `SetOption112` enabled will look like this:
```
tele/tasmota-XXXXXX/Hall_switch/Light/SENSOR
tele/tasmota-XXXXXX/Hall_switch/Fan/SENSOR
tele/tasmota-XXXXXX/Hall_switch/Garage_Door/SENSOR
```

The `SENSOR` json with `SetOption83` and `SetOption100` enabled, will look like this:
```
{"Hall_switch":{"Device":"0x0000","Name":"Light","Power":1,"Endpoint":1,"LinkQuality":99}}
{"Hall_switch":{"Device":"0x0000","Name":"Fan","Power":1,"Endpoint":2,"LinkQuality":99}}
{"Hall_switch":{"Device":"0x0000","Name":"Garage_Door","Power":1,"Endpoint":3,"LinkQuality":99}}
```

For me it makes sense and I am looking forward to hear your opinion too.

**Comment by s-hadinger** [comment]
Good suggestion, that's not a trivial change but I will look forward to it.

**Comment by s-hadinger** [comment]
This feature is now supported by #16709
